agent/discovery_chain_endpoint_test.go

```diff
@@ -188,6 +189,11 @@ func TestDiscoveryChainRead(t *testing.T) {
 				Kind:           structs.ServiceResolver,
 				Name:           "web",
 				ConnectTimeout: 33 * time.Second,
+				Failover: map[string]structs.ServiceResolverFailover{
```

Comment:
After the compiler patch above the gateway customization is now correctly ignored so this test wasn't testing one of the customizations. Adding a dc2 failover causes it to continue being customized by gateway mode.